### DISCUSSION TITLE
What happen when the title of the video has spaces or another symbol

### DISCUSSION THREAD

**Discussion by carloscuervo** [norkunas/youtube-dl-php#244] [General]
If you want to download a video  named as: "KAROL G, Peso Pluma - QLONA (Visualizer)"
Ex: https://www.youtube.com/watch?v=BeUOBoSPWvA

When object video is created, the property filename has characters in the filename that will perform an error when  trying to create an HTTP response with this file because the filename is not normalized.

Is there any way to   normalize this file name?

**Comment by carloscuervo** [comment]
I'm trying restrictFilenames option to solve this

**Reply by norkunas** [reply]
I suggest to avoid using filenames from source and use template to generate by id or something and then store somewhere the original name to display or providing file download with that name.. What's the point of storing with original filename in your filesystem?

**Reply by carloscuervo** [reply]
Thanks for your reply.
I'm developing a website to let people to convert videos, or playlists from YT to audio mp3.
I would like that the name of the file processed (mp3 file) could be the same, or the most similar  to the original one, that's the reason to store them with the original name.
I agree you and I will be store them identified by an id and then mantain a relationship between those ids and the original name.

Thanks!
